error: [Prusti: verification error] postcondition might not hold.
 --> $DIR/bool.rs:6:22
  |
6 | fn test1(b: bool) -> bool {
  |                      ^^^^
  |
note: the error originates here
 --> $DIR/bool.rs:6:1
  |
6 | / fn test1(b: bool) -> bool {
7 | |     !b
8 | | }
  | |_^
note: counterexample for "b"
       value:   true
 --> $DIR/bool.rs:6:10
  |
6 | fn test1(b: bool) -> bool {
  |          ^
note: counterexample for "result"
       value:   false
 --> $DIR/bool.rs:7:5
  |
7 |     !b
  |     ^^

error: [Prusti: verification error] postcondition might not hold.
  --> $DIR/bool.rs:13:22
   |
13 | fn test2(b: bool) -> bool {
   |                      ^^^^
   |
note: the error originates here
  --> $DIR/bool.rs:13:1
   |
13 | / fn test2(b: bool) -> bool {
14 | |     !b
15 | | }
   | |_^
note: counterexample for "b"
       value:   true
  --> $DIR/bool.rs:13:10
   |
13 | fn test2(b: bool) -> bool {
   |          ^
note: counterexample for "result"
       value:   false
  --> $DIR/bool.rs:14:5
   |
14 |     !b
   |     ^^

error: [Prusti: verification error] the asserted expression might not hold
  --> $DIR/bool.rs:18:5
   |
18 |     assert!(b);
   |     ^^^^^^^^^^
   |
note: the failing assertion is here
  --> $DIR/bool.rs:18:5
   |
18 |     assert!(b);
   |     ^^^^^^^^^^
note: counterexample for "b"
       value:   false
  --> $DIR/bool.rs:17:10
   |
17 | fn test3(b: bool) {
   |          ^
   = note: this error originates in the macro `assert` (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 3 previous errors

